SCF/Assignment3A/StringManipulation.java

```diff
@@ -0,0 +1,179 @@
+import java.io.IOException;
+
+/**
+ * 
+ * @author Nilesh Patel
+ * This class designed for provide methods of string manipulation.
+ * This class contain stringComparator, getReverse, replaceLowerCaseWithUpperCase and getLargestWord methods.
+ */
+public class StringManipulation 
+{	
+	/**
+	 * 
+	 * @param firstString : first string which we want to compare.
+	 * @param secondString : second string which we want to compare.
+	 * @fisrtString or @secondString should not be null.
+	 * @return 1 if both string are equal else 0. 
+	 */
+	public int stringComparator(String firstString, String secondString) throws IOException
+	{
+		/*
+		 * Input Validation.
+		 */
+		if(firstString == null || secondString == null)
+		{
+			throw new IOException("Invalid input! found Null string ");
+		}
+		
+		if(firstString.length() != secondString.length())
+		{
+			return 0;
+		}
+		else
+		{
+			/*
+			 * comparing both strings by one by one character. 
+			 */
+			for (int charIndex = 0; charIndex < firstString.length(); charIndex++)
+			{
+				if(firstString.charAt(charIndex) != secondString.charAt(charIndex))
+				{
+					return 0;
+				}
+			}
+			return 1;
+		}
+	}
+	
+	/**
+	 * 
+	 * @param stringToBeReversed : string we want to reverse.
+	 * @return reverse of @stringToBeReversed.
+	 * @stringToBeReversed should not be null;
+	 */
+	public String getReverse(String stringToBeReversed) throws IOException
+	{
+		/*
+		 * Input Validation.
+		 */
+		if(stringToBeReversed == null)
+		{
+			throw new IOException("Invalid input! found Null string ");
+		}
+		
+		/**
+		 * @reversedString will store reverse string.
+		 */
+		StringBuilder reversedString = new StringBuilder();
+		
+		/**
+		 * Traverse string form last index to start index.  
+		 */
+		for (int charIndex = stringToBeReversed.length() - 1; charIndex >= 0; charIndex--) 
+		{
+			reversedString.append(stringToBeReversed.charAt(charIndex));
+		}
+		
+		/**
+		 * making @reversedString immutable.
+		 */
+		String finalreversedString = reversedString.toString();
```

Comment:
merge the line ```80``` and ```81```  to ```return reversedString.toString();``` . Do the same for other methods as well.

Reply:
Done